pkg/daemon/update.go

```diff
@@ -1494,12 +1501,20 @@ func (dn *Daemon) writeUnits(units []ign3types.Unit) error {
 					return err
 				}
 			}
-			// write the unit to disk
 			if err := writeFileAtomicallyWithDefaults(fpath, []byte(*u.Contents)); err != nil {
 				return fmt.Errorf("failed to write systemd unit %q: %v", u.Name, err)
 			}
 
 			glog.V(2).Infof("Successfully wrote systemd unit %q: ", u.Name)
+		} else if u.Mask != nil && !*u.Mask {
+			// if mask is explicitly set to false, make sure to remove a previous mask
+			// see https://bugzilla.redhat.com/show_bug.cgi?id=1966445
+			// note that this does not catch all cleanup cases; for example, if the previous machine config specified Contents, and the current one does not, the previous content will not get cleaned up. For now we're ignoring some of those edge cases rather than introducing more complexity.
+			glog.V(2).Info("Ensuring systemd unit is not masked")
```

Comment:
Might be easier for debugging if we added unit path as well here in the log line. (Side note, I believe V(2) is the default, so I think we can just make it glog.infof)

Reply:
Everything else in the method uses `V(2)`, do you want me to make it inconsistent?

Reply:
Actually I guess it's a mix

Reply:
Yeah I think we were never that good on the consistency front. Not a major issue either way